vivekrnv/sonic-buildimage
# [nvidia][hsflowd] Fix Dropmon co-operation issues related to HW stop

#### Why I did it

Sflow service stop is causing other dropmon clients to not receive any drops thereafter,

**Repro Steps:**

```
config feature state sflow disabled 
config sflow enable
config sflow collector add temp1 192.168.1.10 
	
<Start other dropmon clients>
<Client will be receiving the drops >

config sflow disable <Exit hsflowd process>	
<Other client will stop recieving drops>
```

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

- During process exit, don't stop HW in Drop Mon. for hsflowd. HW Drops are controlled by other daemon in nvidia platform. 
- As for SW drops, only start sw drops in NET_DM when the sw=on is provided in hsflowd.conf
- Don't log feedcontrolerrors for CONFIG since if feedcontrolerrors > 0, application won't stop sw drops even when it exits. CONFIG can likely fail with -EBUSY if the NET_DM is already configured by another daemon.

#### How to verify it

1) Verify the steps and see if the client is receiving the drops, default sw=off case. 
```
Aug 19 00:32:44.212065 r-leopard-41 NOTICE sflow#sflowmgrd: :- sflowHandleService: Starting hsflowd service
Aug 19 00:32:44.212237 r-leopard-41 INFO sflow#hsflowd: started
Aug 19 00:32:44.212282 r-leopard-41 INFO sflow#hsflowd: autoload SONIC and PSAMPLE modules
Aug 19 00:32:44.212282 r-leopard-41 INFO sflow#hsflowd: drop-monitor support for SONiC
Aug 19 00:33:40.446770 r-leopard-41 INFO sflow#hsflowd: dropmon state INIT -> GET_FAMILY
Aug 19 00:33:40.446770 r-leopard-41 INFO sflow#hsflowd: dropmon state GET_FAMILY -> GOT_GROUP
Aug 19 00:33:41.037737 r-leopard-41 INFO sflow#hsflowd: dropmon state GOT_GROUP -> JOIN_GROUP
Aug 19 00:33:42.042191 r-leopard-41 INFO sflow#hsflowd: dropmon state JOIN_GROUP -> CONFIGURE
Aug 19 00:33:42.042191 r-leopard-41 INFO sflow#hsflowd: Configuring DropMon Failed, Module is already in Monitoring State, Continue...
Aug 19 00:33:42.042191 r-leopard-41 INFO sflow#hsflowd: message repeated 2 times: [ Configuring DropMon Failed, Module is already in Monitoring State, Continue...]
Aug 19 00:33:42.042191 r-leopard-41 INFO sflow#hsflowd: dropmon state CONFIGURE -> START
Aug 19 00:34:42.042191 r-leopard-41 INFO sflow#hsflowd: dropmon state START -> RUN
Aug 19 00:35:07.746616 r-leopard-41 INFO sflow#hsflowd: Received SIGTERM
Aug 19 00:35:07.445474 r-leopard-41 INFO sflow#hsflowd: dropmon state RUN -> STOP
Aug 19 00:35:07.795261 r-leopard-41 INFO sflow#hsflowd: stopped
```

2) Test the sw=on case
```
Aug 19 00:55:29.802990 r-leopard-41 INFO sflow#hsflowd: started
Aug 19 00:55:29.802990 r-leopard-41 INFO sflow#hsflowd: autoload SONIC and PSAMPLE modules
Aug 19 00:55:29.802990 r-leopard-41 INFO sflow#hsflowd: drop-monitor support for SONiC
Aug 19 00:55:38.213225 r-leopard-41 INFO sflow#hsflowd: dropmon state INIT -> GET_FAMILY
Aug 19 00:55:38.213225 r-leopard-41 INFO sflow#hsflowd: dropmon state GET_FAMILY -> GOT_GROUP
Aug 19 00:55:38.804221 r-leopard-41 INFO sflow#hsflowd: dropmon state GOT_GROUP -> JOIN_GROUP
Aug 19 00:55:39.808749 r-leopard-41 INFO sflow#hsflowd: dropmon state JOIN_GROUP -> CONFIGURE
Aug 19 00:55:39.808749 r-leopard-41 INFO sflow#hsflowd: Configuring DropMon Failed, Module is already in Monitoring State, Continue...
Aug 19 00:55:40.813554 r-leopard-41 INFO sflow#hsflowd: message repeated 2 times: [ Configuring DropMon Failed, Module is already in Monitoring State, Continue...]
Aug 19 00:55:40.813554 r-leopard-41 INFO sflow#hsflowd: dropmon state CONFIGURE -> START
Aug 19 00:55:42.430937 r-leopard-41 INFO sflow#hsflowd: dropmon state START -> RUN
Aug 19 00:56:36.441267 r-leopard-41 INFO sflow#hsflowd: Received SIGTERM
Aug 19 00:56:36.445474 r-leopard-41 INFO sflow#hsflowd: dropmon: graceful shutdown: turning off feed
Aug 19 00:56:36.445474 r-leopard-41 INFO sflow#hsflowd: dropmon state RUN -> STOP
Aug 19 00:56:36.507932 r-leopard-41 INFO sflow#hsflowd: stopped
```





#### Which release branch to backport (provide reason below if selected)




#### Tested branch (Please provide the tested image version)




#### Description for the changelog




#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
